### ISSUE TITLE
Implement proper attention system for words/phrases similar to AUTOMATIC1111

### ISSUE THREAD

**Issue by n00mkrad** [invoke-ai/InvokeAI#881]
**Is your feature request related to a problem? Please describe.**
Putting emphasis on a single word or phrase does not work properly. Using the prompt weighting syntax described in the InvokeAI docs produces bad results, as the increased weight seems to ignore the rest of the prompt more.
AUTOMATIC1111's solution works much more reliably and is easier to type.

**Describe the solution you'd like**
Implement the syntax/weighting system from AUTOMATIC1111:
https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Features#attentionemphasis

**Describe alternatives you've considered**
I've tried prompt weighting but it does not work great. For example, `blue tree` and `blue:1 tree:1` produce different results even though weight should not change.

**Comment by lstein** [comment]
@blessedcoolant  I basically agree that the current prompt weighting syntax leaves something to be desired.  Is this something you’d like to work on after the GUI work?

**Comment by tildebyte** [comment]
> increased weight seems to ignore the rest of the prompt more.

How would this be handled otherwise, logically? If everything starts out "weight 1" and then one thing is changed to "weight 2", **_by definition_** everything else is "down-weighted"...

Some thoughts:

- prompting is not unlike programming
- programming is accomplished through **extremely** tightly-designed programming languages
- willy-nilly throwing features into the prompting syntax (language) is an **extremely** bad idea

TL;DR we should have a very in-depth discussion about what the prompting language is going to be, and how to design it to be extensible without breaking everything every time a new feature comes in (e.g. we currently have #473 as a feature request, and I recently opened PR #884, *both* of which will require changes/extensions to the prompting language).

**Comment by blessedcoolant** [comment]
> I basically agree that the current prompt weighting syntax leaves something to be desired. Is this something you’d like to work on after the GUI work?

I agree. The GUI work might take quite a bit of time because once the main feature is live, I'll be working on the image2img, inpainting, outpainting UI work .. all of which is pretty time consuming.

I'll try to see if I can have a look at this in between if I get tired of the UI work. Maybe if someone else is willing to pick it up, I can collaborate when needed.

**Comment by lstein** [comment]
Given the choice between the web GUI and redesigning the prompt  language, you should go for the GUI!  Is there a way of integrating a sketch pad into the GUI? I frequently want to erase a hand that went wrong, and it is a pain to go into a photo editor just for that one step.

**Comment by blessedcoolant** [comment]
> Is there a way of integrating a sketch pad into the GUI?

Yes. I was discussing the idea today with @psychedelicious on how we want to implement the canvas. Because inpainting and outpainting have very similar functionality, once we have that planned, we can build out the canvas and the editor system for a good UI experience.

**Comment by psychedelicious** [comment]
I've started a discussion about the syntax: #904

**Comment by lstein** [comment]
Superseded by #904
